### ISSUE TITLE
Ghost pours?

### ISSUE THREAD

**Issue by Boombrewer** [RaspberryPints/RaspberryPints#293]
Is anyone else getting the occasional 'ghost' pour?  I've got a few kegs that were getting close to empty and then all of the sudden, they show "MT" and it looks as though pours were completed (i.e. - system is working fine).  I had this problem at one point during initial testing and I was getting 'ghost' signals from my sensors due to a loose connection.  Not sure if that is an issue again (everything seems snug) or if anyone else is seeing this.

Also, I was going to just delete the pour from the table to correct the volume but from what I can tell reading the code (stop laughing) once the pour is registered and 'amountRemaining' is updated, it doesn't use the pour table again.  So I'd have to manually adjust the appropriate fields in the vwGetTapsAmountPoured table if I want to correct my display - is that correct?

**Comment by mcangeli** [comment]
No. You can remove the bad pour from the pours table. 

The two tables with vw are views,  created on the fly when they are called with data from multiple tables.

I've noticed ghost pours when there is any chance of air in the lines.

Sent via the Samsung GALAXY S®4, an AT&T 4G LTE smartphone

<div>-------- Original message --------</div><div>From: Boombrewer notifications@github.com </div><div>Date:09/17/2014  9:33 PM  (GMT-05:00) </div><div>To: RaspberryPints/RaspberryPints RaspberryPints@noreply.github.com </div><div>Subject: [RaspberryPints] Ghost pours? (#293) </div><div>
</div>Is anyone else getting the occasional 'ghost' pour? I've got a few kegs that were getting close to empty and then all of the sudden, they show "MT" and it looks as though pours were completed (i.e. - system is working fine). I had this problem at one point during initial testing and I was getting 'ghost' signals from my sensors due to a loose connection. Not sure if that is an issue again (everything seems snug) or if anyone else is seeing this.

Also, I was going to just delete the pour from the table to correct the volume but from what I can tell reading the code (stop laughing) once the pour is registered and 'amountRemaining' is updated, it doesn't use the pour table again. So I'd have to manually adjust the appropriate fields in the vwGetTapsAmountPoured table if I want to correct my display - is that correct?

**Comment by Boombrewer** [comment]
I think it was a combination of tension on the lead (from the new enclosure) and power connected to the RPi instead of the Alamode.  But in messing around with it, I corrupted the SD.  So I'll be starting over... some other night.  

Thanks for the help though.

**Comment by day-trippr** [comment]
I had a nasty ghost pour problem that was eventually resolved when I finally swapped out the meter with my spare. Looking at the log file I've kept it was coughing up single ounces, many ounces, and hundreds of ounces, and the frequency was increasing rapidly. I moved it to another keg - thus different electrical and plumbing circuits - and at the end it was failing within minutes of resetting the keg. Since I replaced it a week ago I have not had a single unexplained ounce across all six taps.

I truly am quite pleased about the whole thing.

As for "starting over":  Ugh. I felt your pain once. Then I learned to backup my SD card whenever I accomplish anything significant (and given I knew zilch about Debian, LXDE, Python, SQLite3 or MySQL, and barely could stand on my hind legs with html and javascript, pretty much anything that works is significant ;) ).

Win32DiskImager on Windows, or DD on the 'Pi, to an attached card reader, and you can do all kinds of brave/silly/downright dumb things to your system - and get it all back again in minutes. 

Between building RPi systems, installing and up-revving RPints and BrewPi and writing my own stuff, I couldn't begin to count how many times I've dumb-thumbed something in the last 9 months that would have put me right back to Start without the collection of SD card images I have on my workstation...

Cheers! ;)

**Comment by Boombrewer** [comment]
Yeah, I was hoping it wasn't the sensor but wasn't ruling it out either.  Just hoping I don't have to go through that work.

And yeah, I learned the clone thing myself.  I have a clone from August so I just have to remember what stage it was at.  I think I'll just be able to restore that image, do an 'update & upgrade', and install the new version of RPints.  So not too bad.

But I'm also having power issues through the alamode.  So I'm guessing I'll be replacing that as well.  If I plug power into the alamode, the RPi won't boot up.  The power light comes on but it won't boot.  Not sure if I shorted something or not.  Luckily, I think I have another alamode that was used for another project.
